Module 3 tada create param ref and tada create use param ref updates

wokenny13 opened the pull request:
This PR focuses on the updates that have been made to TADA_CreateParamRef and TADA_CreateUseParamRef

(Next pull request will focus on the additional Spatial crosswalks and TADA_DefineCriteriaMethodology functions. Excluded those functions from this PR.)

In this PR please look through:

- TADA Module 3a vignette updates
- Review the new arg input for TADA_CreateParamRef(fillBy = "None", "Org" and "All)
- Review the new arg input for TADA_CreateUseParamRef(auto_assign = TRUE/FALSE)
- R documentation for these two functions and the examples
- The excel file export option and if there are any issues with the flagging columns, usability, dropdown lists, etc.

wokenny13 committed:
Update CriteriaMethods.R

This creates DefineMagnitude in which is the dataframe users will define Magnitude Criteria components

wokenny13 committed:
DefineMagnitude Updates

Fixed some formatting with the DefineMagnitude output table and other Continued developments.

wokenny13 committed:
Updated DefineMagnitude()

DefineMagnitude format has been updated. Minor bug fix in TADA_CreateParamRef() was fixed.

wokenny13 committed:
updates from first mod 3 functions pr

wokenny13 committed:
Update CST.csv

wokenny13 committed:
MagnitudeSummary (roughdraft)

need to review logic in summary output and handling of empty column cases and how to best summarize impairments

wokenny13 committed:
Magnitude Excrusion summary

wokenny13 committed:
Merge branch 'Create-Module-3-Reference-Tables' into Create-Module-3-AURef

wokenny13 committed:
Update CriteriaMethods.R

wokenny13 committed:
Create CriteriaPlot.R

wokenny13 committed:
Delete CriteriaPlot.R

wokenny13 committed:
Create CriteriaPlot.R

wokenny13 committed:
Update from Mod 3 Current PR

wokenny13 committed:
Draft AU to Use Ref functions

Define Magnitude Excursion Summary

wokenny13 committed:
Updates to AU-ML-USE crosswalk

Summary of magnitude excursions edits with formatting and logic are also included in this commit.

wokenny13 committed:
Update TADAModule3.Rmd

wokenny13 committed:
Updates to Mod 3 workflows Vignette

wokenny13 committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Update TADA_CreateUseParamRef

Renamed paramUseRef argument to useParamRef

wokenny13 committed:
Update Module 3 Functions

Removed duplicate TADA functions that were moved around from different branches.

wokenny13 committed:
Update ATTAINSCrosswalks

wokenny13 committed:
Minor Text Updates

wokenny13 committed:
Scratch Work TADA_CreateUseAURef

Adding WaterType column and consider renaming of function.

wokenny13 committed:
Update TADA_CreateUseAURef

New required argument inputs. WaterType column will be included in this output. ATTAINS.ParameterName column added as well.

Users will need to specify unique spatial criteria for a Param-Use-WaterType during this function.

wokenny13 committed:
Minor Text Changes

wokenny13 committed:
Rename and Update TADA_CreateWaterUseParamRef

wokenny13 committed:
Updates

wokenny13 committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Updates to Geospatial function

chunk size and blank space in URL handling

wokenny13 committed:
Update Geospatial function

wokenny13 committed:
Update ATTAINSCrosswalks.R

Editing documentation

Updates to 2 Spatial crosswalks:
TADA_CreateUseAURef
TADA_CreateWaterUseParamRef

wokenny13 committed:
Updates to TADA_CreateSpatialRef

committing current progress with updates to this function.

wokenny13 committed:
Edits to TADA Spatial

Mod3b vignette and function updates

wokenny13 committed:
Update TADAModule3b.Rmd

wokenny13 committed:
Updates to the Spatial Crosswalk Tables

Removed CriteriaMethods.R for now. This file will generate the final DefineCriteriaMethods table for users to fill out.

Documentation updates, TADA Module3b vignette updates, and other styling push

wokenny13 committed:
Created Example data frame

Data_Nutrients_UT_GetATTAINS contains output of our Data_Nutrients_UT when we run TADA_GetATTAINS.

wokenny13 committed:
Update All TADA Ref Files

Ran TADA_Update functions to update ref tables

wokenny13 committed:
Capitalization Argument Input Fix

lowercase 1st letter of an argument input

wokenny13 committed:
Documentation Update

wokenny13 committed:
DESCRIPTION and 3b Vignette Update

Return table outputs in Mod 3b vignette to view in vignette

Added rExpertQuery Github to DESCRIPTION.

wokenny13 committed:
Update DESCRIPTION

Typo fix

wokenny13 committed:
Update DESCRIPTION

wokenny13 committed:
Loading Example Data Issue

Attempted to store an example data of TADA_GetATTAINS(Data_Nutrients_UT) in the EPATADA Package.

Ran into an issue with updating example data.

For now, I will run the TADA_GetATTAINS() directly in the example data frame for the R documentation examples. Runtime may be around 5 minutes.

wokenny13 committed:
Updates

hillarymarler committed:
update wordlist and global vars

hillarymarler committed:
fix data set name issue

hillarymarler committed:
ex data set fixes

hillarymarler committed:
Add TADA's rExpertQuery API key

hillarymarler committed:
update ex data

wokenny13 committed:
Function documentation

added and removed param argument documentation that was missing

global variables fix - added missing ones to utilities. Note: I need to consider which column names to keep in the return data frame of each Mod 3 functions.

Mod 3 vignette update.

wokenny13 committed:
Updates

wokenny13 committed:
vignette update and mod 3 function updates

cleaned up code, made mod 3 functions more consistent with formatting and coding logic.

made org_id a required input.

Handling of cases of making the crosswalk tables reusable.

vignette workflow incorporated.

wokenny13 committed:
mod 3 Reusable Crosswalk table input Updates

Update logic on how to flag/identify new, existing and dropped row combinations of values and how to label them.

wokenny13 committed:
updates to mod 3 functions

IncludeOrExclude columns for ref tables

wokenny13 committed:
Reorganize (arrange) rows in outputs

make include rows show up on top.

reformat crosswalk table outputs.

mod 3 vignette modification

wokenny13 committed:
text edits

wokenny13 committed:
Document updates and spell checks

document updates

mod 3b vignette updates

spell checks/wordlist updates

wokenny13 committed:
Update TADAModule3b.Rmd

changed argument inputs in vignette to excel =FALSE

wokenny13 committed:
Merge develop branch

wokenny13 committed:
conflict fix in GeospatialFunctions.R

wokenny13 committed:
Required Maintainenance Run

Edit conflicts and ran maintenance

wokenny13 committed:
Update Global Variables

update global variables list and removed intermediate variables

Created/updated Data_Nutrients_GetATTAINS.rda

wokenny13 committed:
Mod 3 function updates

Improved TADA_ParamRef flagging logic.

Partial push of DefineCriteriaMethodology

wokenny13 committed:
Excluding TADA_DefineMagnitude

Still in development, commenting out for now

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

cristinamullin committed:
Merge branch 'geospatialrmd' into Create-Module-3-AURef

wokenny13 committed:
Update Utilities.R

Missing comma in global variables list

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Temporary push for 5/12 Subgroup

Reflects TADA_DefineMagnitude (in development still - r documentation is still needed as well as some additional review and fixes to this function)

wokenny13 committed:
Push TADA_DefineMagnitude function

Updated documentation for TADA_DefineMagnitude function - ready for review.

Additional excel formatting updates to TADA_CreateSpatialRef

wokenny13 committed:
Updated example data frames

cristinamullin committed:
update TADAModule3a

wokenny13 committed:
CreateParam and UseParam updates

Removing EPA304a.PollutantName column.

wokenny13 committed:
Remove addition of EPA304a standards

Focusing on crosswalk ATTAINS and WQP for now.

wokenny13 committed:
Updates to mod 3 function

standardized column names when pulling in ATTAINS data and creating Mod 3 Crosswalk reference tables

updated vignettes

updated flag logic

wokenny13 committed:
Update test-Utilities.R

wokenny13 committed:
Renamed and Updates to Mod 3A vignette

wokenny13 committed:
updates

wokenny13 committed:
Updates to TADA_CreateParamRef and TADA_CreateUseParamRef

argument inputs created to assist with auto populating entries for exact matches.

allows for further specification if an org only wants org specific exact matches

wokenny13 committed:
Updates to TADA_CreateParamRef and TADA_CreateUseParamRef

wokenny13 committed:
Spatial Ref update

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
Updates to TADA_SpatialRef

wokenny13 committed:
Update documentation

For TADA_CreateParamRef and TADA_CreateUseParamRef

wokenny13 committed:
Merge branch 'Create-Module-3-AURef' of https://github.com/USEPA/EPATADA into Create-Module-3-AURef

wokenny13 committed:
Update ATTAINSCrosswalks.R

Updated documentation

wokenny13 committed:
function name change, bug fix and documentation updates

Updates to reflect changes to TADA_CreateParamRef()

documentation updates, bug fixes, and changed TADA_DefineMagnitude to TADA_DefineCriteriaMethodology

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

cristinamullin committed:
maintenance and TADA_GetATTAINSParamUseOrgRef bug fix

remove commented out code no longer needed with rExpertQuery update. TADA_UpdateATTAINSParamUseOrgRef is working now. Added TADA_UpdateATTAINSParameterWQPCharRef to required maintenance file. Updated all ref files and example data. Updated spelling list. Ran styler

cristinamullin committed:
Update CriteriaMethods.R

cristinamullin committed:
Update TADA_DefineCriteriaMethodology.Rd

wokenny13 committed:
Update to TADA_CreateParamRef and Removal of a TADA_DefineCriteriaMethodology arg input

Was not capturing the entire Index of allowable values when excel file was generated with TADA_CreateParamRef

Removed an arg input in TADA_DefineCriteriaMethodology for the time being.

wokenny13 committed:
Update ATTAINSCrosswalks.R

Excel CreateParamRef and CreateUseParamRef Flagging function update to reflect correct MATCH columns appropriately.

wokenny13 committed:
Updates

cristinamullin committed:
Merge branch 'develop' into Create-Module-3-AURef

wokenny13 committed:
update flagging columns in TADA_UseParamRef

wokenny13 committed:
Update ATTAINSCrosswalks.R

wokenny13 committed:
TADA_CreateParamRef and TADA_CreateUseParamRef updates

Flagging logic consistant in both Excel and in RStudio output

text fixes

new arg inputs with auto_assign in TADA_CreateUseParamRef and fillBy in TADA_CreateParamRef

wokenny13 committed:
Updates to TAD_CreateParamRef and TADA_CreateUseParamRef

TADA_CreateParamRef:

new fillBy arg input
new flag column to indicate if a row was modified and how - useful for new WQX Char names if a user reuses a crosswalk reference table.

TADACreateUseParamRef: new auto_assign arg input
new flag column to indicate if a row was modified and how - Useful for when new Uses & param combo now exist in ATTAINS if a user reuses a crossswalk reference table.

wokenny13 committed:
Update TADAModule3a.Rmd

Updates to reflect new arg inputs in these 2 functions (TADA_CreateParamRef and TADACreateUseParamRef)

wokenny13 committed:
Update TADAModule3a.Rmd

data frame typo fix.

wokenny13 committed:
Update TADAModule3a.Rmd

cristinamullin commented on DESCRIPTION:
```diff
@@ -77,6 +77,7 @@ Imports:
     jsonlite, 
     openxlsx,
     rATTAINS,
```
Is rATTAINS still needed? It looks like it is only used to get the ATTAINS domain files now (in the files below), is that correct? 

TADA_GetATTAINSAUSiteCrosswalk
TADA_GetATTAINSParameterWQPCharRef
TADA_GetATTAINSOrgIDsRef
TADA_GetATTAINSParamUseOrgRef
TADA_TableExport (example only)
TADAModule3a.Rmd

wokenny13 replied:
I believe we should be able to now leverage just rExpertQuery. The domain_value function within it should be working fine now. I can look to getting this incorporated to remove the rATTAINS dependency in those other functions.

cristinamullin replied:
Awesome! thank you - anytime we can remove dependencies that is really helpful

hillarymarler replied:
I'm out the rest of this week @wokenny13, but if you want to work on replacing any rATTAINS use with rExpertQuery functions, I can help with that next week if that would be helpful.

wokenny13 replied:
I tried to make the update in TADA_GetATTAINSAUSiteCrosswalk consistent.

The number of rows do end up differing where au.crosswalk with 368 obs is from the orginal code with rATTAINS for AK

and the 208 obs is for the one with rExpertQuery::EQ_AUsMLs

Please take a look to review if the logic looks correct when you get the chance @hillarymarler I will push the updates through in this PR
<img width="334" alt="image" src="https://github.com/user-attachments/assets/98e7f3e3-5a6c-4165-a1f9-4db1ebb7dbb6" />

hillarymarler replied:
I will take a look at this today.

hillarymarler replied:
@wokenny13 the difference is that the rATTAINS version is returning all assessment units (active, retired, and historical), while the rExpertQuery version returns only active assessment units.

I think the desired behavior is to return only active assessment units, but we can modify the rExpertQuery version of the query to return retired/historical units too if that is desired. What are your thoughts?

wokenny13 replied:
Only the active ones make sense to me. I thought rATTAINS was only returning the active ones as well, so this explains the difference.

cristinamullin commented on R/ExampleData.R:
```diff
@@ -15,6 +15,20 @@
 #' @format A dataframe with 14592 rows and 152 variables
 NULL
 
+#' Data_Nutrients_UT_GetATTAINS
```
I don't see this used anywhere. Is it used in a vignette?

wokenny13 replied:
This is used in the Rdocumentation example for one of the Mod 3 spatial functions I believe which is not reflected in this PR

cristinamullin replied:
do you want to remove it from this PR and include it in the next one with the spatial functions?

wokenny13 replied:
I can remove it from this PR

cristinamullin commented on R/ExampleData.R:
```diff
@@ -122,6 +136,20 @@ NULL
 #' @format A data frame with 84 rows and 161 variables
 NULL
 
+#' Data_WV_Mod1_Output_GetATTAINS
```
same questions - I don't see this used anywhere. Is it used in a vignette?

wokenny13 replied:
I don't think this was being used anywhere (I will double check in the other branch with the mod 3 spatial functions), so I can also remove this as well in this PR for now. And if i do see it pop up anywhere it can be added back on.

cristinamullin committed:
Update ATTAINSParamUseEntityRef.csv

cristinamullin commented on inst/WORDLIST:
```diff
@@ -65,8 +68,13 @@ CountyCode
 CreateComparableID
 CreatePairRef
```
CreatePairRef and CreateParamRef function names are too similar... let's update the names to better reflect what each function does

wokenny13 committed:
Typo fix

This typo (missing period) caused the fillBy = Org to return a bug.

Fixed this issue.

wokenny13 committed:
Merge branch 'Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates' of https://github.com/USEPA/EPATADA into Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates

cristinamullin committed:
update wordlist

cristinamullin commented on vignettes/images/FlowChartAUModule3.jpg:
Let's try to think of names that are more intuitive for users. This might be a good question for the subgroup.

TADA_ParametersForAnalysis
TADA_UsesForAnalysis
TADA_MonitoringLocationsForAnalysis

cristinamullin replied:
other ideas:

TADA_SelectParameters
TADA_SelectUses
TADA_SelectMonitoringLocations

wokenny13 committed:
Merge branch 'Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates' of https://github.com/USEPA/EPATADA into Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates

wokenny13 committed:
Removed rATTAINS dependency with rExpertQuery

Removed rATTAINS dependency and replaced use cases with rExpertQuery

Removed irrelevant Example dataframes in this PR

Modified Mod3a vignette

cristinamullin committed:
Update TADAModule3a.Rmd

cristinamullin commented on R/ATTAINSCrosswalks.R:
```diff
@@ -565,7 +577,12 @@ TADA_UpdateMonitoringLocationsInATTAINS <- function(org_id = NULL,
+#'
+#' # How does fillBy = "All" compare to paramRef_UT3?
+#' paramRef_UT4 <- TADA_CreateParamRef(Data_Nutrients_UT,
+#'   org_id = "UTAHDWQ", fillBy = "All", excel = FALSE
```
what do you think of calling this function input "autofill" instead of fillBy?

wokenny13 replied:
In TADA_CreateUseParamRef we use the input name as "auto_assign". So not sure if we want to consider keeping these the same or not. I am open to considering different name for these functions and function inputs as were suggested in earlier comments too.

cristinamullin replied:
I agree they should be the same. I like autofill or auto_assign.

wokenny13 committed:
Update mod 3a vignette

explicitly named each dataframe to reflect what each dataframe is showing from each TADA mod 3 function and its arg inputs.

Fixed a small bug in flagging column for TADA_CreateUseParamRef

Removed EPA304a section as this was moved to only be shown in the final CriteriaMethods table (still in draft)

wokenny13 committed:
Update TADAModule3a.Rmd

wokenny13 committed:
Removed Two .rda files

‘Data_Nutrients_UT_GetATTAINS’ ‘Data_WV_Mod1_Output_GetATTAINS’

as these are not relevant to this PR

wokenny13 commented:
I updated the vignette to make the names of the crosswalk example data frames functionality more explicit and fixed some small flagging bugs so that everything should be working as intended.

I kept the function names and argument input the same for now and we can consider more appropriate new names. We can also consider the flagging columns in both functions and see if they can be reworded in any way for better clarity.

This should be in a good spot for a more final review now.

wokenny13 committed:
Excel conditional formatting and minor vignette edit

color coded excel column values

explicit arg input for excel = FALSE throughout

cristinamullin committed:
Merge branch 'develop' into Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates

cristinamullin committed:
Merge branch 'develop' into Module-3-TADA_CreateParamRef-and-TADA_CreateUseParamRef-Updates

cristinamullin committed:
maintenance

cristinamullin committed:
Update TADA_FindNearbySites.Rd

cristinamullin committed:
Update TADAWaterSciConWorkshopDemo.Rmd

cristinamullin committed:
update TADA_CreateParamRef

change fillBy to auto_assign in TADA_CreateParamRef

cristinamullin commented on vignettes/images/FlowChartAUModule3.jpg:
@wokenny13 can you remove the "CWA 303(d) assessments" language here. This should be more generic, for example, "Water Quality Data Analysis Process - WQP Monitoring Location Level" and "Water Quality Data Analysis Process - ATTAINS Assessment Unit Level"

wokenny13 replied:
This has been pushed through

cristinamullin commented:
I changed fillBy to auto_assign in TADA_CreateParamRef for consistency. 

@wokenny13 I added one comment about the figures. Once that is addressed, this can be merged. Great work!!

wokenny13 committed:
Update vignette Mod3 images

Changed title of mod 3 workflow images to WQP 'Analysis'
